packages/block-library/src/post-author/index.php

```diff
@@ -37,15 +39,40 @@ function post_author_build_css_colors( $attributes ) {
 		$text_colors['css_classes'][] = sprintf( 'has-%s-color', $attributes['textColor'] );
 	} elseif ( $has_custom_text_color ) {
 		// Add the custom color inline style.
-		$text_colors['inline_styles'] .= sprintf( 'color: %s;', $attributes['customTextColor'] );
+		$text_colors['inline_styles'] .= sprintf( 'color: %s;', $attributes['style']['color']['text'] );
 	}
 
+	// Link colors.
+	$has_link_color = array_key_exists( 'style', $attributes )
+	&& array_key_exists( 'color', $attributes['style'] )
+	&& array_key_exists( 'link', $attributes['style']['color'] );
+
+	if ( $has_link_color ) {
+		$text_colors['css_classes'][] = 'has-link-color';
+		// If link is a named color.
+		if ( strpos( $attributes['style']['color']['link'], 'var:preset|color|' ) !== false ) {
+			// Get the name from the string and add proper styles.
+			$index_to_splice               = strrpos( $attributes['style']['color']['link'], '|' ) + 1;
+			$link_color_name               = substr( $attributes['style']['color']['link'], $index_to_splice );
+			$text_colors['inline_styles'] .= sprintf( '--wp--style--color--link:var(--wp--preset--color--%s);', $link_color_name );
+		} else {
+			$text_colors['inline_styles'] .= sprintf( '--wp--style--color--link: %s;', $attributes['style']['color']['link'] );
+		}
+	}
 	// Background color.
 	$has_named_background_color  = array_key_exists( 'backgroundColor', $attributes );
-	$has_custom_background_color = array_key_exists( 'customBackgroundColor', $attributes );
+	$has_custom_background_color = array_key_exists( 'style', $attributes )
+	&& array_key_exists( 'color', $attributes['style'] )
+	&& array_key_exists( 'background', $attributes['style']['color'] );
+
+	// Gradient color.
+	$has_named_gradient  = array_key_exists( 'gradient', $attributes );
+	$has_custom_gradient = array_key_exists( 'style', $attributes )
+	&& array_key_exists( 'color', $attributes['style'] )
+	&& array_key_exists( 'gradient', $attributes['style']['color'] );
 
 	// If has background color.
-	if ( $has_custom_background_color || $has_named_background_color ) {
+	if ( $has_custom_background_color || $has_named_background_color || $has_named_gradient || $has_custom_gradient ) {
 		// Add has-background-color class.
 		$background_colors['css_classes'][] = 'has-background-color';
```

Comment:
Im wondering if this should just be 'has-background'.  I notice that `__experimentalColor` applies the 'has-background' class as opposed to the 'has-background-color' class:
https://github.com/WordPress/gutenberg/blob/c53d36b84282a4668b9b84b0e91b33e175542a27/packages/block-editor/src/hooks/color.js#L130-L133

Im assuming either this should be changed to 'has-background', or the hook should be using 'has-background-color' ?

Reply:
Definitely change this to has-background  we should match `__experimentalColor` 💯